pkg/messaging/rabbitmq/publisher.go

```diff
@@ -51,16 +52,20 @@ func (pub *publisher) Publish(topic string, msg messaging.Message) error {
 	if msg.Subtopic != "" {
 		subject = fmt.Sprintf("%s.%s", subject, msg.Subtopic)
 	}
+	subject = formatTopic(subject)
+
 	err = pub.ch.Publish(
 		exchangeName,
 		subject,
 		false,
 		false,
 		amqp.Publishing{
-			Headers:     amqp.Table{},
-			ContentType: "application/octet-stream",
-			AppId:       "mainflux-publisher",
-			Body:        data,
+			Headers:      amqp.Table{},
+			ContentType:  "application/octet-stream",
+			AppId:        "mainflux-publisher",
+			DeliveryMode: 2,
```

Comment:
What are those magic numbers? These need to be numerical constants, defined at the top of the file with comments explaining their meaning.